### PR TITLE
sharev2: manage replica quotas transparently

### PR DESCRIPTION
**Pull request by majewsky**
@Carthaca Can you double-check that this has the correct semantics?

### CODE REVIEW

**Review by Carthaca — CHANGES_REQUESTED** [review]
I'm not sure if will be safe to enable the `replication_enabled` flag that comes with this change. Please answer my inline questions.

And out of curiosity: is there a reason why you sometimes use `shareTypeName` and in other places `stName`?

File: pkg/plugins/manila.go [hunk 1]
```diff
@@ -64,11 +64,6 @@ func (p *manilaPlugin) Init(provider *gophercloud.ProviderClient, eo gophercloud
 	}
 	p.hasReplicaQuotas = microversion >= 53
 
-	//TODO remove this feature gate from the config once support is fully fleshed out
-	if !p.cfg.ShareV2.HasReplicaQuotas {
-		p.hasReplicaQuotas = false
-	}
-
```
[the remark this answers is not in the document]
Comment (majewsky) [reply pkg/plugins/manila.go:71]: Yes. If no share type has `replication_enabled: true`, Limes should not touch replica quotas and thus behave exactly like before.

File: pkg/plugins/manila.go [hunk 2]
```diff
@@ -225,21 +207,29 @@ func (p *manilaPlugin) Scrape(provider *gophercloud.ProviderClient, eo gopherclo
...
-		result[p.makeResourceName("shares", shareType)] = quotaSets[shareType].Shares.ToResourceData(nil)
-		result[p.makeResourceName("share_snapshots", shareType)] = quotaSets[shareType].Snapshots.ToResourceData(nil)
-		result[p.makeResourceName("share_capacity", shareType)] = quotaSets[shareType].Gigabytes.ToResourceData(gigabytesPhysical)
-		result[p.makeResourceName("snapshot_capacity", shareType)] = quotaSets[shareType].SnapshotGigabytes.ToResourceData(snapshotGigabytesPhysical)
+		sharesData := quotaSets[stName].Shares.ToResourceData(nil)
+		shareCapacityData := quotaSets[stName].Gigabytes.ToResourceData(gigabytesPhysical)
+		if p.hasReplicaQuotas && shareType.ReplicationEnabled {
+			sharesData.Usage = quotaSets[stName].Replicas.Usage
+			shareCapacityData.Usage = quotaSets[stName].ReplicaGigabytes.Usage
+		}
```
[the remark this answers is not in the document]
Comment (majewsky) [reply pkg/plugins/manila.go:227]: Your assessment is correct. Deploying Limes is rather straightforward, though, so we should be able to work this out in practice.

**Comment by majewsky** [comment]
> And out of curiosity: is there a reason why you sometimes use `shareTypeName` and in other places `stName`?

I initially had `shareTypeName` everywhere, but it's quite bulky, so I shortened it to `stName` inside all loops over share types, where the surrounding code provides sufficient context to infer that "st" is "share type".

**Review by Carthaca — APPROVED** [review]

**Pull request merged**